### PR TITLE
Add orderId param to getTrades method

### PR DESCRIPTION
**Pull request by ramirotw**
The orderId param was left behind and it was never added to the getTrades method, to [simplify things](https://github.com/cowprotocol/explorer/blob/35-epic-home-page/src/api/operator/operatorApi.ts#L161) on the Explorer we need this change

Would fix https://github.com/cowprotocol/explorer/issues/118

### CODE REVIEW

File: src/types/utilities.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+type UnionKeys<T> = T extends T ? keyof T : never
+type StrictUnionHelper<T, TAll> = T extends any ? T & Partial<Record<Exclude<UnionKeys<TAll>, keyof T>, never>> : never
+export type StrictUnion<T> = StrictUnionHelper<T, T>
```
Comment (alfetopito) [inline src/types/utilities.d.ts:3]: Can you do anything about this warning?

**Review by matextrem — APPROVED** [review]

File: src/types/utilities.d.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+type UnionKeys<T> = T extends T ? keyof T : never
+type StrictUnionHelper<T, TAll> = T extends unknown
+  ? T & Partial<Record<Exclude<UnionKeys<TAll>, keyof T>, never>>
+  : never
```
Comment (anxolin) [inline src/types/utilities.d.ts:4]: 🤯 I have to say, i didn't get how this work. 
I see it does though..

Comment (W3stside) [reply src/types/utilities.d.ts:4]: Partial and Exclude AND UnionKeys 😂 yeah i have trouble reading this but if it works...? 🤷

File: src/api/cow/cow.spec.ts [hunk 3]
```diff
@@ -265,6 +265,35 @@ test('Valid: Get last 5 trades for a given trader ', async () => {
...
+    `https://api.cow.fi/rinkeby/api/v1/trades?orderUid=${TRADE_RESPONSE.orderUid}&limit=5`,
+    FETCH_RESPONSE_PARAMETERS
+  )
+  expect(trades.length).toEqual(5)
+})
+
+test('Invalid: Get trades passing both the owner and orderId', async () => {
+  await expect(
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```
Comment (W3stside) [inline src/api/cow/cow.spec.ts:287]: what's the reason for this/error?

Comment (W3stside) [reply src/api/cow/cow.spec.ts:287]: oh i see it's the `owner and ownerID` can also use
```ts
@ts-expect-error - <reason string>
```

File: src/api/cow/index.ts [hunk 4]
```diff
@@ -127,10 +127,13 @@ export class CowApi {
   }
 
   async getTrades(params: GetTradesParams): Promise<TradeMetaData[]> {
-    const { owner, limit, offset } = params
-    const qsParams = objectToQueryString({ owner, limit, offset })
+    const { owner, orderId, limit, offset } = params
+    if (owner && orderId) {
```
Comment (W3stside) [inline src/api/cow/index.ts:131]: is it maybe better to create a union type for the params?
sth like:
```ts
type GetTradeParams = OrderParamsWithOrderId | OrderParamsWithOwner
```

Comment (W3stside) [reply src/api/cow/index.ts:131]: haha ok i see it happened below, sorry

**Review by W3stside — APPROVED** [review]

File: src/types/utilities.d.ts (hunk 2 above)
Comment (ramirotw) [reply src/types/utilities.d.ts:4]: yeah... it would be great if TS already implemented this as a utility type

**Pull request merged**